flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/serialization/SpillingAdaptiveSpanningRecordDeserializer.java

```diff
@@ -101,11 +99,13 @@ public DeserializationResult getNextRecord(T target) throws IOException {
 	}
 
 	private DeserializationResult readNonSpanningRecord(T target) throws IOException {
-		NextRecordResponse response = nonSpanningWrapper.getNextRecord(target);
-		if (response.result == PARTIAL_RECORD) {
-			spanningWrapper.transferFrom(nonSpanningWrapper, response.bytesLeft);
+		int recordLen = nonSpanningWrapper.readInt();
+		if (nonSpanningWrapper.canReadRecord(recordLen)) {
+			return nonSpanningWrapper.readInto(target);
```

Comment:
Leave a comment that this can't be turned into POJO/Tuple because of performance?